src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1933,13 +1982,24 @@ public void decreaseOverallHappiness(final double amount)
 
     /**
      * Get all the waypoints of the colony.
+     *
      * @return copy of hashmap.
      */
     public Map<BlockPos, IBlockState> getWayPoints()
     {
         return new HashMap<>(wayPoints);
     }
 
+    public List<EntityLivingBase> getGuardTargets()
+    {
+        return new ArrayList<>(guardTargets);
+    }
+
+    public void setGuardTargets(final List<EntityLivingBase> targets)
+    {
+        guardTargets = targets;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "targets". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)